test/unit/firestore-collection.js

```diff
@@ -356,4 +356,31 @@ describe('MockFirestoreCollection', function () {
       ]);
     });
   });
+  describe('#listDocuments', function () {
+    it('retrieves all data for existing collection', function(done) {
+      db.autoFlush();
+      var keys = Object.keys(require('./data.json').collections);
+      collection.listDocuments().then(function(refs) {
+        expect(refs.length).to.equal(6);
+        refs.forEach(function(ref) {
+          expect(keys).to.contain(ref.id);
+        });
+        done();
+      }).catch(done);
```

Comment:
Isn't this going to silently swallow exceptions that occur before any assertions are run and give us a false pass?

Reply:
Good question. Mocha's `done` accepts an error argument: https://mochajs.org/#asynchronous-code